backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java

```diff
@@ -136,6 +148,29 @@ public TestResultUpload processResultCSV(InputStream csvStream) {
     return csvResult;
   }
 
+  private byte[] translateSpecimenNameToSNOMED(byte[] content, Map<String, String> snomedMap) {
+    String[] rows = new String(content, StandardCharsets.UTF_8).split("\n");
+    String headers = rows[0];
+
+    int specimenTypeIndex =
+        Arrays.stream(headers.split(",")).toList().indexOf(SPECIMEN_TYPE_COLUMN_NAME);
+
+    for (int i = 1; i < rows.length; i++) {
+      var row = rows[i].split(",", -1);
+      var specimenTypeName = Arrays.stream(row).toList().get(specimenTypeIndex).toLowerCase();
+
+      if (!specimenTypeName.matches(ALPHABET_REGEX)) {
+        continue;
+      }
+
+      row[specimenTypeIndex] = snomedMap.get(specimenTypeName);
```

Comment:
so if it's not in the map is this value set to `null`?  Or does it throw an exception?

Reply:
It will be `null`. The punchline will be the same regardless of how we handle this: if nothing else, ReportStream will see the missing or invalid specimen type value and reject the submission. We could be a little more explicit about handling this, though - might be a good opportunity to give the user a better error.

Reply:
Per @zdeveloper's suggestion, the validation step will now recognize all valid specimen names and return an error on any invalid name ([0842531](https://github.com/CDCgov/prime-simplereport/pull/5670/commits/0842531db56e8591c19e254b5c842a7971d71ac0)).

So, by the time we get to the editing step, it is guaranteed that the provided name will exist in the map and this will never be `null`.